src/vs/platform/terminal/node/terminalEnvironment.ts

```diff
@@ -87,14 +88,33 @@ export async function getShellIntegrationInjection(
 	}
 
 	const originalArgs = shellLaunchConfig.args;
-	const shell = process.platform === 'win32' ? path.basename(shellLaunchConfig.executable).toLowerCase() : path.basename(shellLaunchConfig.executable);
+	let shell = process.platform === 'win32' ? path.basename(shellLaunchConfig.executable).toLowerCase() : path.basename(shellLaunchConfig.executable);
 	const appRoot = path.dirname(FileAccess.asFileUri('').fsPath);
 	const type = 'injection';
 	let newArgs: string[] | undefined;
 	const envMixin: IProcessEnvironment = {
 		'VSCODE_INJECTION': '1'
 	};
 
+	async function checkIfShIsActuallyBash(): Promise<boolean> {
+		const versionString = await execShVersion();
+		const bashRegex = /\bbash\b/i;
+
+		if (!versionString) {
+			return false;
+		}
+		if (bashRegex.test(versionString)) {
+			envMixin['VSCODE_SH_IS_BASH'] = '1';
+			return true;
+		}
+
+		return false;
+	}
+
+	if (envMixin['VSCODE_SH_IS_BASH'] === '1' || (shell === 'sh' && await checkIfShIsActuallyBash())) {
+		shell = 'bash';
```

Comment:
We successfully change from Sh to Bash here.
However, still no shell integration